tools/cldr-code/src/main/java/org/unicode/cldr/util/MatchValue.java

```diff
@@ -108,6 +114,37 @@ public static MatchValue of(String command) {
         }
     }
 
+    public static class BCP47LocaleMatchValue extends MatchValue {
```

Comment:
You should at least check for validity of the LSRV components using the calls as LocaleMatchValue; see 

            lang = new ValidityMatchValue(LstrType.language, statuses, false);

and following.


For the extensions, I checked, and it turns out that we would need to do a bit of work to pull some pieces together to do a full validity check. So that can be a follow-on.

Reply:
I'll file a ticket for the extension validity checking

Reply:
@macchiati this checks for well-formed bcp47. If we check for valid bcp47, it would not be forwards-compatible with codes assigned that this version of CLDR doesn't know about.

Perhaps in the future we could add more parameters to switch validity on as well?

Reply:
We update each version of CLDR, and the chances of someone wanting to submit a keyboard for a language between the time that ISO approves a language code and CLDR releases is very small. And a validity check will help prevent malformed locale IDs, a recurrent problem.

Reply:
That's true, but tooling from CLDR version, say, v44 _could_ be used to validate a keyboard for a language code not assigned until 2025. So the DTD should not be defined in terms of "bcp47 codes valid at the time of CLDR release".

Reply:
Perhaps what we should have is a separate check, that verifies that all keyboards _in CLDR_ have valid codes. That would be separate from passing a DTD `@MATCH` test

Reply:
@macchiati  I've added a subtask, https://unicode-org.atlassian.net/browse/CLDR-16950 to enforce valid locale IDs or keyboards which are (or are contemplating being!) included in CLDR's common data. 

Can we merge this or perhaps rename to bcp47-wellformed or something?

Reply:
That works.

Reply:
updated to `@MATCH validity/bcp47-wellformed`